src/main/java/javax/measure/unit/ProductUnit.java

```diff
@@ -188,6 +188,9 @@ static Unit<? extends Quantity> getProductInstance(Unit<?> left, Unit<?> right)
      */
     static Unit<? extends Quantity> getQuotientInstance(Unit<?> left, Unit<?> right) {
         Element[] leftElems;
+        if (left == right) {
```

Comment:
We should check equals, not identity.

Reply:
Checking identity here should be good enough, right? since we now cache instances of `Unit` subclasses.

Reply:
We only cache ProductUnit, none of the others. But as mentioned further in the PR, this doesn't solve the issue and there's a better solution.